File: JavaScript/hw4/Vector.js
class Vector{
    constructor(v){
        this.dim = v.length;
        this.comp = v;
    }

    dimension(){
        return this.dim;
    }

    component(opts = {}){
        if(!opts['index'])
            return this.comp;
        return this.comp[opts['index']-1];
    }

    length(){
        var s = 0;
        for(var i in this.comp)
            s += this.comp[i] * this.comp[i];
        return Math.sqrt(s);
    }

}
function getVector(s){
    if((s.charAt(0) !== '(') || (s.charAt(s.length - 1) !== ')'))
        throw new Error('Format Error');
    s = s.substring(1, s.length - 1);
    s = s.split(',');
    v = [];
    for(i = 0; i < s.length; i++)
        v.push(parseInt(s[i]));
    return new Vector(v);
}

File: Ruby/hw4/Vector.rb
class Vector
    def initialize(v)
        if (v.length < 1)
            raise ArgumentError, "Dimension Error"
        end
        @comp = v
        @d = v.length
    end

    def component(n = 0)
        if (n == 0)
            return @comp
        end
            @comp[n - 1]
    end

    def dimension
        @d
    end

    def length
        s = 0
        for i in @comp
            s += i **2
        end
        s ** 0.5
    end

end

def getVector(s)
    if ((s[0] != '(') || (s[-1] != ')'))
        raise ArgumentError, "Format Error"
    end
    s = s[1..-1]
    s = s.split(',')
    v = []
    for i in s
        v.push(Decimal(i))
    end
    Vector.new(v)
end
